### PR TITLE
fix: Avoid IE problems by using nodeName instead of tagName

### PR DESCRIPTION
**Pull request by jkodu**
This is a maintenance PR, to update `node.tagName` usages to `node.nodeName`.

Did not update the usages in:
- `lib/core/utils/css-parser.js` - as this is a copy/pasted (imported) module.
- `lib/core/utils/qsa.js` - as this seems to break a lot of tests.

Closes issue:
- https://github.com/dequelabs/axe-core/issues/942

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Has documentation updated, a DU ticket, or requires no documentation change
- [x] Includes new tests, or was unnecessary
- [x] Code is reviewed for security by: << Marcy Sutton >>

### CODE REVIEW

File: lib/checks/navigation/heading-order.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ if (ariaHeadingLevel !== null) {
 	return true;
 }
 
-var headingLevel = node.tagName.match(/H(\d)/);
+var headingLevel = node.nodeName.match(/H(\d)/);
```
Comment (marcysutton) [inline lib/checks/navigation/heading-order.js]: do you need `toUpperCase()` here?

**Comment by marcysutton** [comment]
I'm seeing additional usage in `commons/color/get-background-color`, should that be updated as well?

File: lib/checks/navigation/heading-order.js (hunk 1 above)
Comment (jkodu) [reply lib/checks/navigation/heading-order.js]: I am going to leave this on as is (no need to uppercase).

**Comment by jkodu** [comment]
@marcysutton - the usage in `commons/color/get-background-color`, seems to me like an object with property `tagName` rather than an `HTMLElement` itself. Hence did not change it (as doing so broke a lot of tests).

File: lib/checks/navigation/heading-order.js (hunk 1 above)
Comment (WilcoFiers) [reply lib/checks/navigation/heading-order.js]: I disagree. In xhtml nodeName is lower case, which means this won't match.

**Pull request merged**